sonar-db/src/main/java/org/sonar/db/DatabaseChecker.java

```diff
@@ -43,12 +44,29 @@ public DatabaseChecker(Database db) {
   @Override
   public void start() {
     try {
-      if (H2.ID.equals(db.getDialect().getId())) {
-        Loggers.get(DatabaseChecker.class).warn("H2 database should be used for evaluation purpose only");
-      } else if (Oracle.ID.equals(db.getDialect().getId())) {
-        checkOracleVersion();
+      switch (db.getDialect().getId()) {
+        case H2.ID:
+          Loggers.get(DatabaseChecker.class).warn("H2 database should be used for evaluation purpose only");
+          break;
+        case Oracle.ID:
+          checkOracleDriverVersion();
+          checkMinDatabaseVersion(db.getDialect(), 11, 0);
+          break;
+        case MySql.ID:
+          checkMinDatabaseVersion(db.getDialect(), 5, 6);
+          break;
+        case PostgreSql.ID:
+          checkMinDatabaseVersion(db.getDialect(), 8, 0);
+          break;
+        case MsSql.ID:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1151)